core/src/main/java/org/apache/carbondata/core/datamap/DataMapStoreManager.java

```diff
@@ -322,6 +322,18 @@ public TableDataMap getDataMap(CarbonTable table, DataMapSchema dataMapSchema) {
         tableIndices = allDataMaps.get(tableUniqueName);
       }
     }
+    // in case of fileformat, when table is dropped the datamaps are not cleared, they need to be
+    // cleared by using API, so in the same session, if the table is created again with same name
+    // and columns(may be with different name or datatype), the datamap will be returned for old
+    // table only as the table unique name is same, so compare the columns, if not same, clear the
+    // datamaps on that table
+    if (allDataMaps.size() > 0 && null != allDataMaps.get(tableUniqueName)
+        && allDataMaps.get(tableUniqueName).size() > 0 && !allDataMaps.get(tableUniqueName).get(0)
+        .getTable().getTableInfo().getFactTable().getListOfColumns()
+        .equals(table.getTableInfo().getFactTable().getListOfColumns())) {
```

Comment:
No need to check this list of columns equals check. Just check for tableid would be enough

Reply:
this is actually checking when the schema is updated. So im checking for columns and then removing the datamap, and creates new datamap for the latest schema

Reply:
Just check for tableid would be enough